Feature/ #636 Module Specific Readme Files

10upsimon opened the pull request:
## Summary

Addresses #636 

## Relevant technical choices

- Adds a draft v1 `readme.txt` file to the `webp-uploads` module folder for copying over to the relevant build directory as per the work carried out in #662 .
- Flags all `readme.txt` files in all performance plugin module folders for `export-ignore` via `.gitattributes` glob pattern match.

## Checklist

- [x] PR has either `[Focus]` or `Infrastructure` label.
- [x] PR has a `[Type]` label.
- [x] PR has a milestone or the `no milestone` label.

10upsimon committed:
Add readme.txt placeholder files to all wpp modules. Content to be added in later commits.

10upsimon committed:
Draft v1 of webp-uploads module standalone readme

10upsimon committed:
Added placeholder readme.txt file to sqlite module folder

10upsimon committed:
Added placeholder readme.txt file to audit autoload options module folder

10upsimon committed:
Add export-ignore flags for all standalone module readme.txt files

10upsimon committed:
Setting version of webp-uploads plugin readme file to 1.0.0

10upsimon committed:
Removed readme.txt files from modules that will not be unbundled

felixarntz commented on .gitattributes:
```diff
@@ -30,3 +30,11 @@
+/modules/images/dominant-color/readme.txt export-ignore
+/modules/images/fetchpriority/readme.txt export-ignore
+/modules/images/webp-support/readme.txt export-ignore
+/modules/images/webp-uploads/readme.txt export-ignore
```
The requirements state to only add a readme for WebP Uploads as part of this issue. Let's stick to that and ignore all other modules.

10upsimon replied:
@felixarntz  I've changed this to a single 1 line glob pattern now, based on @joemcgill 's feedback so that we can match current and future `readme.txt` files

```
/modules/**/readme.txt export-ignore
```

Let me know if this is good, or if you want me to specifically define it as webp-uploads only, and single out modules in the future.

felixarntz reviewed (commented):
@10upsimon Can you please limit the changes here to just the WebP Uploads module? Other modules to deploy as standalone plugins will be handled in a separate issue.

10upsimon committed:
Removed all readme.txt files from modules other than webp-uploads

10upsimon committed:
Added glob pattern to match all readme.txt files in module folders in order to export-ignore them

10upsimon commented:
@felixarntz I've addressed your comment RE restricting this to `webp-uploads` only, however I did implement a catch-all glob pattern match for all module `readme.txt` files in order to `export-ignore` them.

Once @mukeshpanchal27 's PR at #662 is approved and merged, I'll enhance the CLI command to handle the copy of the `readme.txt` files.

felixarntz commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,63 @@
+License URI:       https://www.gnu.org/licenses/gpl-2.0.html
+Tags:              performance, images, site health, measurement, webp
+
+WebP uploads module for the [performance plugin](https://wordpress.org/plugins/performance-lab/) from the WordPress Performance Team.
```
I don't think we should mention the Performance Lab plugin here given that this is the description for the standalone plugin. I think we can use the description that is also present in the module header:
```suggestion
Creates WebP versions for new JPEG image uploads if supported by the server.
```

felixarntz commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,63 @@
+
+== Description ==
+
+This plugin adds a WebP uploads module to the [performance plugin](https://wordpress.org/plugins/performance-lab/), enabling WebP support for media uploads within the WordPress application, when this module is enabled.
+
+This plugin is intended to be used alongside the [performance plugin](https://wordpress.org/plugins/performance-lab/), and will not function as a standalone plugin. The performance plugin may be obtained [here](https://wordpress.org/plugins/performance-lab/).
```
The latter is not correct, this plugin _will_ function as a standalone plugin. That's part of why we're breaking them apart.

I would avoid the mention of the Performance Lab plugin here and focus purely on a brief description of what this plugin does.

felixarntz commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,63 @@
+=== Performance Lab - WebP Uploads Module ===
```
The plugin should be called like the module, "WebP Uploads". It does not require Performance Lab, so shouldn't look as if it was an extension.

felixarntz commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,63 @@
+
+== Installation ==
+
+= Installation from within WordPress =
+
+1. Visit **Plugins > Add New**.
+2. Search for **Performance Lab - WebP Uploads Module**.
+3. Install and activate the **Performance Lab - WebP Uploads Module** plugin.
+
+= Manual installation =
+
+1. Upload the entire `performance-lab-webp-support-module` folder to the `/wp-content/plugins/` directory.
+2. Visit **Plugins**.
+3. Activate the **Performance Lab - WebP Uploads Module** plugin.
```
See above:
```suggestion
= Installation from within WordPress =

1. Visit **Plugins > Add New**.
2. Search for **WebP Uploads**.
3. Install and activate the **WebP Uploads** plugin.

= Manual installation =

1. Upload the entire `webp-uploads` folder to the `/wp-content/plugins/` directory.
2. Visit **Plugins**.
3. Activate the **WebP Uploads** plugin.
```

felixarntz commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,63 @@
+2. Visit **Plugins**.
+3. Activate the **Performance Lab - WebP Uploads Module** plugin.
+
+= After activation =
+
+1. Visit the new **Settings > Performance** menu.
+2. Enable the individual modules you would like to use, one of which should be this WebP support module.
```
This should be removed, as the plugin will work as soon as it's activated.

felixarntz commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,63 @@
+
+== Frequently Asked Questions ==
+
+= What is the purpose of this plugin? =
+
+The primary purpose of the Performance Lab - WebP Support Module plugin is to add an additional module to the performance plugin that enables the ability to toggle on/off WebP support.
+
```
I think we can remove this.

felixarntz commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,63 @@
+
+= Can I use this plugin on my production site? =
+
+Per the primary purpose of the plugin (see above), it can be considered stable for production usage given that this particular module is not marked as experimental.
```
This refers to a module, which it shouldn't. The standalone plugin will have no notion of what is experimental or not.

felixarntz commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,63 @@
+
+= Where can I submit my plugin feedback? =
+
+Feedback is encouraged and much appreciated, especially since this plugin may contain future WordPress core features. If you have suggestions or requests for new features, you can [submit them as an issue in the Performance Lab GitHub repository](https://github.com/WordPress/performance/issues/new/choose). If you need help with troubleshooting or have a question about the plugin, please [create a ...
```
```suggestion
Feedback is encouraged and much appreciated, especially since this plugin may contain future WordPress core features. If you have suggestions or requests for new features, you can [submit them as an issue in the WordPress Performance Team's GitHub repository](https://github.com/WordPress/performance/issues/new/choose). If you need help with troubleshooting or have a question about the plugin, please [create a new topic on our support forum](https://wordpress.org/support/plugin/webp-uploads/#new-topic-0).
```

felixarntz commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,63 @@
+= How can I contribute to the plugin? =
+
+Contributions are always welcome! Learn more about how to get involved in the [Core Performance Team Handbook](https://make.wordpress.org/performance/handbook/get-involved/).
+
```
Can you add (slightly modified versions of) the two WebP Uploads specific entries in our current Performance Lab `readme.txt` here? Those are relevant and should be part of this readme. The mentioning of activating the module in Performance Lab should be removed from them though, as just by activating this plugin on its own those two questions may be relevant.

felixarntz commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,63 @@
+
+**Initial Release**
+
+* Initial release of the WebP support performance plugin module as a standalone plugin. ([636](https://github.com/WordPress/performance/issues/636))
```
We typically refer to PRs in the changelog instead of issues.
```suggestion
* Initial release of the WebP Uploads plugin as a standalone plugin. ([664](https://github.com/WordPress/performance/pull/664))
```

felixarntz reviewed (commented):
@10upsimon Thank you for the updates. I think there is a misunderstanding here, as the standalone plugins will be in fact standalone. They should not be branded/marketed as extensions to the Performance Lab plugin.

Right now, as part of Milestone 1 (see #656), they will exist in both places, as a standalone plugin and as a module in the Performance Lab plugin. But once we're done with Milestone 2, they will only exist as standalone plugin.

Using Performance Lab alongside them is perfectly fine and we would still encourage users to use Performance Lab if they want to follow along what the performance team is working on. But someone that wants to just use WebP would be perfectly fine to only install this plugin  on its own.

Regarding the dependency on #662, I don't think we technically have to wait for that PR. Once this PR is finalized, it can be merged as is, since #662 will copy the entire directory already, so the `readme.txt` here will be included.

mukeshpanchal27 commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,63 @@
+Stable tag:        1.0.0
+License:           GPLv2 or later
+License URI:       https://www.gnu.org/licenses/gpl-2.0.html
+Tags:              performance, images, site health, measurement, webp
```
```suggestion
Tags:              performance, images, webp
```

I don't think we have to add `site health` and `measurement` for WebP Uploads

mukeshpanchal27 reviewed (commented):
Thanks @10upsimon, I left one minor suggestion, and could you please remove the `readme.txt` file for sqlite? 

10upsimon committed:
Merge branch 'feature/creating-standalone-plugins' of github.com:WordPress/performance into feature/creating-standalone-plugins

10upsimon committed:
Merge branch 'feature/creating-standalone-plugins' into add/issue-636-module-specific-readme-files

10upsimon committed:
Remove dangling readme.txt file from SQLite module

10upsimon committed:
Ongoing modifications to WebP Uploads readme.txt following review feedback

10upsimon committed:
Ongoing modifications to WebP Uploads readme.txt following review feedback

10upsimon commented:
@mukeshpanchal27 @felixarntz many thanks for your feedback and comments. I believe I have addressed them all and have requested a re-review of the PR. I confirm the readme.txt was copied across as expected with @mukeshpanchal27 latest merge of #662  @felixarntz

10upsimon committed:
Amended readme.txt plugin description text for webp-uploads plugin

mukeshpanchal27 commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,60 @@
+
+This plugin adds WebP support for media uploads within the WordPress application.
+
+
```
```suggestion
```
Remove extra line.

mukeshpanchal27 reviewed (approved):
Thanks @10upsimon, The changes look good to me.

felixarntz committed:
Remove unnecessary extra line.

Co-authored-by: Mukesh Panchal <mukeshpanchal27@users.noreply.github.com>

felixarntz commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,59 @@
+
+== Description ==
+
+This plugin adds WebP support for media uploads within the WordPress application.
```
Technically not a blocker, but since this is pretty much the only module / plugin that has a UI to it, we still may want to provide at least a bit more explanation in 2-3 short paragraphs. Things to potentially highlight:
* WebP images will be generated only for new uploads
* By default, only WebP images will be generated for JPEG uploads, only the original uploaded file will still exist as JPEG
* There is a checkbox in `Settings > Media` that can change the behavior though to generate _both_ JPEG and WebP for every sub-size (noting again that this will only affect newly uploaded images, i.e. after making that change)

10upsimon replied:
@felixarntz (cc @mukeshpanchal27 ) this is ready for re-review following changes pushed in recent commit(s).

felixarntz commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,59 @@
+
+== Changelog ==
+
+= 1.0.0 =
+
+**Initial Release**
```
No need for an extra "heading" here
```suggestion
= 1.0.0 =
```

felixarntz reviewed (commented):
@10upsimon This is pretty much good to go now, just one small thing and one slightly bigger thing. It would be great if you could expand the description just a tiny bit to explain a bit more clearly what the plugin does.

We can also iterate in a follow up if you prefer for just that content, in order to unblock this PR. Let me know what you prefer.

mukeshpanchal27 committed:
Update modules/images/webp-uploads/readme.txt

Co-authored-by: Felix Arntz <felixarntz@users.noreply.github.com>

10upsimon committed:
Merge branch 'add/issue-636-module-specific-readme-files' of github.com:WordPress/performance into add/issue-636-module-specific-readme-files

10upsimon committed:
Enhancements to webp-uploads readme.txt file following review

mukeshpanchal27 commented on modules/images/webp-uploads/readme.txt:
```diff
@@ -0,0 +1,57 @@
+
+== Description ==
+
+This plugin adds WebP support for media uploads within the WordPress application. WebP images will be generated only for new uploads, preexisting imagery will not be converted to WebP format. By default, WebP images will only be generated for JPEG uploads, only the original uploaded file will still exist as a JPEG image. ALl generated image sizes will exist as WebP only. If you wish to change thi ...
```
```suggestion
This plugin adds WebP support for media uploads within the WordPress application. WebP images will be generated only for new uploads, pre-existing imagery will not be converted to WebP format. By default, WebP images will only be generated for JPEG uploads, only the original uploaded file will still exist as a JPEG image. All generated image sizes will exist as WebP only. If you wish to change this behaviour, there is a checkbox in `Settings > Media` that - when checked - will alter the behaviour of this plugin to generate both JPEG and WebP images for every sub-size (noting again that this will only affect newly uploaded images, i.e. after making said change).
```

10upsimon replied:
Thanks @mukeshpanchal27 , updated

10upsimon committed:
Enhancements to webp-uploads readme.txt file following review

felixarntz reviewed (approved):
@10upsimon Thank you for the updates, LGTM!
